src/SourceBuild/content/Directory.Build.targets

```diff
@@ -1,16 +1,32 @@
 <Project>
 
-  <Target Name="DetermineMicrosoftSourceBuildIntermediateInstallerVersion">
-    <!-- Manually load the installer version from the PVP.     -->
-    <XmlPeek XmlInputPath="$(IntermediatePath)PackageVersions.package-source-build.Current.props"
-             Query="msb:Project/msb:PropertyGroup/msb:MicrosoftSourceBuildIntermediateInstallerVersion/text()"
-             Namespaces="&lt;Namespace Prefix='msb' Uri='http://schemas.microsoft.com/developer/msbuild/2003'/&gt;">
-        <Output TaskParameter="Result" ItemName="MicrosoftSourceBuildIntermediateInstallerVersionItem" />
+  <Target Name="DetermineSourceBuiltSdkVersion">
+    <ItemGroup>
+      <SdkTarballItem Include="$(SourceBuiltAssetsDir)dotnet-sdk-*$(TarBallExtension)" />
+    </ItemGroup>
+    <PropertyGroup>
+      <SdkTarball>%(SdkTarballItem.Identity)</SdkTarball>
+      <SdkLayout>$(ArtifactsTmpDir)Sdk</SdkLayout>
+      <BundledVersionsPropsRelativePath>sdk/*/Microsoft.NETCoreSdk.BundledVersions.props</BundledVersionsPropsRelativePath>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(SdkLayout)" />
+    <Exec Command="tar -xzf $(SdkTarball) -C $(SdkLayout)"
```

Comment:
Extracting the entire tarball feels a bit heavy handed.  Can the version simply be parsed out from the SDK tarball itself?

Reply:
Thanks - yest, it should be rather easy to parse the SDK tarball name. Will fix.

> Extracting the entire tarball feels a bit heavy handed. Can the version simply be parsed out from the SDK tarball itself?

Thanks - yes, it should be rather easy to parse the SDK tarball name. Will fix.

Reply:
If necessary, you can also just extract the .version file from the tarball as well.

Reply:
I did try that first, but single file extraction didn't work on some distros (Alpine?) - so I quickly reverted to full tarball extraction, which is an overkill.

Reply:
Fixed with https://github.com/dotnet/installer/pull/18153/commits/35d1fad7115150eb3b9352a918dd0263eca42524